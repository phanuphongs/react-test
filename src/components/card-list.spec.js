import { shallow } from 'enzyme';
import React from 'react';
import CardList from "./card-list";

it('expect to render Card List component', () => {
    const mockData = [
        {
            id: 1,
            name: 'John Walker',
            email: 'johnwalker@gmail.com',
        }
    ]
    expect(shallow(<CardList robots={mockData}/>)).toMatchSnapshot( )
})
