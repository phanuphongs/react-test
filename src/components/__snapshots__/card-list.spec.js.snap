// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card List component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "johnwalker@gmail.com",
              "id": 1,
              "name": "John Walker",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          robot={
                    Object {
                              "email": "johnwalker@gmail.com",
                              "id": 1,
                              "name": "John Walker",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "robot": Object {
            "email": "johnwalker@gmail.com",
            "id": 1,
            "name": "John Walker",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            robot={
                        Object {
                                    "email": "johnwalker@gmail.com",
                                    "id": 1,
                                    "name": "John Walker",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "robot": Object {
              "email": "johnwalker@gmail.com",
              "id": 1,
              "name": "John Walker",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`expect to render Card component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "johnwalker@gmail.com",
              "id": 1,
              "name": "John Walker",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          robot={
                    Object {
                              "email": "johnwalker@gmail.com",
                              "id": 1,
                              "name": "John Walker",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "robot": Object {
            "email": "johnwalker@gmail.com",
            "id": 1,
            "name": "John Walker",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            robot={
                        Object {
                                    "email": "johnwalker@gmail.com",
                                    "id": 1,
                                    "name": "John Walker",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "robot": Object {
              "email": "johnwalker@gmail.com",
              "id": 1,
              "name": "John Walker",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
